public/app/core/services/backend_srv.ts

```diff
@@ -142,6 +142,56 @@ export class BackendSrv implements BackendService {
     });
   }
 
+  chunked(options: BackendSrvRequest): Observable<FetchResponse<Uint8Array | undefined>> {
+    const controller = new AbortController();
+    const requestId = options.requestId ?? `chunked-${Date.now()}`;
+    return new Observable((observer) => {
+      let disconnect = false;
+      const sub = this.fetch<ReadableStream<Uint8Array>>({
+        ...options,
+        requestId,
+        responseType: 'body',
+        showErrorAlert: false,
+        showSuccessAlert: false,
+        hideFromInspector: true,
+        abortSignal: controller.signal,
+      }).subscribe((result) => {
+        const reader = result.data.getReader();
+        async function process() {
+          while (!disconnect) {
+            try {
+              const chunk = await reader.read();
+              observer.next({
+                ...result,
+                data: chunk.value,
+              });
+              if (chunk.done) {
+                break;
+              }
+            } catch (err) {
+              console.warn('error while reading chunked stream', options.url, err);
+              reader.cancel('error');
+              observer.error(err);
+            }
+          }
+          if (disconnect) {
+            // this happens when we unsubscribe before the connection is done
+            // Catch the error because it may be locked from the abort controller
+            result.data.cancel('disconnect').catch(() => {});
```

Comment:
🤯  @Clarity-89  any thoughs on how to better disconnect?  This works for me locally -- eg, see the screenshot at the top; each refresh stops loading the previous request

BUT when I test on the ephemeral instance, it does not stop:
https://ephemeral1511182198691ryantxu.grafana-dev.net/d/de9gh5kbp3x8ga/streaming-example

![2025-01-09_21-53-51 (1)](https://github.com/user-attachments/assets/7bebb0eb-b30e-4f81-99d3-c366414350ba)

Reply:
Hard to say here without knowing why it doesn't work properly on cloud. Could it be that we also need to call `reader.cancel` if disconnect is true?